Framework/DataHandling/src/LoadILLDiffraction.cpp

```diff
@@ -264,14 +268,93 @@ void LoadILLDiffraction::initStaticWorkspace() {
  * Use the ScanningWorkspaceBuilder to create a time indexed workspace.
  *
  * @param scan : scan data
+ * @param start_time : start time in ISO format string
  */
-void LoadILLDiffraction::initMovingWorkspace(const NXDouble &scan) {
+void LoadILLDiffraction::initMovingWorkspace(const NXDouble &scan,
+                                             const std::string &start_time) {
   const size_t nTimeIndexes = m_numberScanPoints;
   const size_t nBins = 1;
   const bool isPointData = true;
 
-  const auto instrumentWorkspace = loadEmptyInstrument();
+  const auto instrumentWorkspace = loadEmptyInstrument(start_time);
   const auto &instrument = instrumentWorkspace->getInstrument();
+  auto &params = instrumentWorkspace->instrumentParameters();
+
+  const auto &referenceComponentPosition =
+      getReferenceComponentPosition(instrumentWorkspace);
+
+  double refR, refTheta, refPhi;
+  referenceComponentPosition.getSpherical(refR, refTheta, refPhi);
+
+  if (m_instName == "D2B") {
+    auto &compInfo = instrumentWorkspace->mutableComponentInfo();
+
+    Geometry::IComponent_const_sptr detectors =
+        instrument->getComponentByName("detectors");
+    const auto detCompIndex = compInfo.indexOf(detectors->getComponentID());
+    const auto tubes = compInfo.children(detCompIndex);
+    const size_t nTubes = tubes.size();
+    Geometry::IComponent_const_sptr tube1 =
+        instrument->getComponentByName("tube_1");
+    const auto tube1CompIndex = compInfo.indexOf(tube1->getComponentID());
+    const auto pixels = compInfo.children(tube1CompIndex);
+    const size_t nPixels = pixels.size();
+
+    Geometry::IComponent_const_sptr pixel =
+        instrument->getComponentByName("standard_pixel");
+    Geometry::BoundingBox bb;
+    pixel->getBoundingBox(bb);
+    m_pixelHeight = bb.yMax() - bb.yMin();
+
+    const auto tubeAnglesStr = params.getString("D2B", "tube_angles");
```

Comment:
I am not sure putting the moving based on the parameters file here is a good idea. For any non-moving D2B file these corrections will not be applied, even though the loader docs say they will be. They will also not be applied when doing load empty instrument. I think it would be best to put these corrections in the IDF - they can be repeated here if the instrument scientists want the information, or put as a comment in the IDF.